mongodbatlas/cloud_provider_snapshot_export_buckets.go

```diff
@@ -0,0 +1,163 @@
+		return nil, nil, err
+	}
+
+	root := new(CloudProviderSnapshotExportBucket)
+	resp, err := c.Client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+// Create Allows you to grant Atlas access to the specified bucket for exporting backup snapshots.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/cloud-backup/export/create-one-export-bucket/
+func (c CloudProviderSnapshotExportBucketsServiceOp) Create(ctx context.Context, projectID string, bucket *CloudProviderSnapshotExportBucket) (*CloudProviderSnapshotExportBucket, *Response, error) {
+	if projectID == "" {
+		return nil, nil, NewArgError("projectID", "must be set")
+	}
+
+	path := fmt.Sprintf(cloudProviderSnapshotExportBucketsPath, projectID)
+
+	req, err := c.Client.NewRequest(ctx, http.MethodPost, path, bucket)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(CloudProviderSnapshotExportBucket)
+	resp, err := c.Client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+// Delete Allows you to remove one bucket specified by the bucket ID.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/cloud-backup/export/delete-one-export-bucket/
+func (c CloudProviderSnapshotExportBucketsServiceOp) Delete(ctx context.Context, projectID, bucketID string) (*Response, error) {
+	if projectID == "" {
+		return nil, NewArgError("projectID", "must be set")
+	}
+	if bucketID == "" {
+		return nil, NewArgError("bucketID", "must be set")
+	}
+
+	path := fmt.Sprintf("api/atlas/v1.0/groups/%s/backup/exportBuckets/%s", projectID, bucketID)
+
+	req, err := c.Client.NewRequest(ctx, http.MethodDelete, path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.Client.Do(ctx, req, nil)
+
+	return resp, err
+}
+
+var _ CloudProviderSnapshotExportBucketsService = &CloudProviderSnapshotExportBucketsServiceOp{}
+
+// CloudProviderSnapshotExportBuckets represents all cloud provider snapshot export buckets.
+type CloudProviderSnapshotExportBuckets struct {
+	Results    []*CloudProviderSnapshotExportBucket `json:"results,omitempty"`    // Includes one CloudProviderSnapshotExportBucket object for each item detailed in the results array section.
+	Links      []*Link                              `json:"links,omitempty"`      // One or more links to sub-resources and/or related resources.
+	TotalCount int                                  `json:"totalCount,omitempty"` // Count of the total number of items in the result set. It may be greater than the number of objects in the results array if the entire result set is paginated.
+}
+
+// CloudProviderSnapshotExportBucket represents one cloud provider snapshot export buckets.
+type CloudProviderSnapshotExportBucket struct {
+	ID            string `json:"_id,omitempty"`           // Unique identifier of the S3 bucket.
+	ProjectID     string `json:"projectId,omitempty"`     // Unique 24-hexadecimal digit string identifying the project.
```

Comment:
[q] this is only documented for the create, but even so the example doesn't use it, is this really necessary?

Reply:
I can remove it, but I don't know if the end users will need it, due it's in the create documentation.

Reply:
if you have tested the `curl`s and it does actually show or is needed we keep it, I'm just wondering if this is a documentation error, not a blocker to keep